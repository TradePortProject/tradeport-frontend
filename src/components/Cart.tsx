import React from "react";
import { useSelector, useDispatch } from "react-redux";
import {
  removeFromCart,
  updateQuantity,
  increaseQuantity,
  addToCart,
} from "../store/features/cartSlice";
import { RootState, AppDispatch } from "../store/store";

const Cart: React.FC = () => {
  const cartItems = useSelector((state: RootState) => state.cart.items);
  const dispatch = useDispatch<AppDispatch>();
  const cartItemCount = useSelector((state: RootState) =>
    state.cart.items.reduce((total, item) => total + item.quantity, 0),
  );

  const handleRemoveFromCart = (productId: number) => {
    dispatch(removeFromCart(productId));
  };

  const handleIncreaseQuantity = (productId: number) => {
    dispatch(increaseQuantity(productId));
  };

  const handleQuantityChange = (productId: number, quantity: number) => {
    dispatch(updateQuantity({ productId, quantity }));
  };

  const handleAddToCart = (product: {
    id: number;
    name: string;
    price: number;
  }) => {
    dispatch(addToCart(product));
  };

  return (
    <div>
      <h2>Shopping Cart</h2>
      <p>Total Items: {cartItemCount}</p>

      <button
        className="bg-red-300"
        onClick={() => {
          handleAddToCart({ id: 1, name: "Sample Product", price: 10 });
        }}
      >
        Add Sample Product
      </button>

      {cartItems.length === 0 ? (
        <p>Your cart is empty.</p>
      ) : (
        <ul>
          {cartItems.map((item) => (
            <li key={item.product.id}>
              {item.product.name} - Quantity: {item.quantity}
              <button
                className="mx-4 rounded-lg bg-red-600 p-2 text-white"
                onClick={() => handleRemoveFromCart(item.product.id)}
              >
                Remove
              </button>
              <button onClick={() => handleIncreaseQuantity(item.product.id)}>
                +
              </button>
              <input
                type="number"
                value={item.quantity}
                onChange={(e) =>
                  handleQuantityChange(
                    item.product.id,
                    parseInt(e.target.value, 10),
                  )
                }
              />
            </li>
          ))}
        </ul>
      )}
    </div>
  );
};

export default Cart;
