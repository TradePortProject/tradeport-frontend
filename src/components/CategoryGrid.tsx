import React from "react";

interface CategoryProps {
  image: string;
  title: string;
}

const categories: CategoryProps[] = [
  {
    image: "/img/sofa.jpg",
    title: "Living Room",
  },
  {
    image: "/img/bedroom.jpg",
    title: "Bedroom",
  },
  {
    image: "/img/kitchen.jpg",
    title: "Kitchen",
  },
  {
    image: "/img/bathroom.jpg",
    title: "Bathroom",
  },
  {
    image: "/img/outdoor.jpg",
    title: "Outdoor",
  },
  {
    image: "/img/office.jpg",
    title: "Office",
  },
];

const CategoryGrid: React.FC = () => {
  return (
    <div className="mx-auto max-w-7xl px-4 py-12">
      <h2 className="mb-8 text-center text-3xl font-bold text-gray-900">
        Shop by Category
      </h2>

      {/* Grid Layout */}
      <div className="grid grid-cols-1 gap-6 px-12 sm:grid-cols-2 md:grid-cols-3 md:px-0">
        {categories.map((category, index) => (
          <div
            key={index}
            className="group relative overflow-hidden rounded-lg shadow-lg"
          >
            {/* Background Image */}
            <img
              src={category.image}
              alt={category.title}
              className="h-48 w-full object-cover transition duration-500 group-hover:brightness-75 md:h-64"
            />

            {/* Text Overlay */}
            <div className="absolute inset-0 flex items-center justify-center bg-black bg-opacity-30 opacity-100 transition-all duration-500 group-hover:bg-opacity-50">
              <h3 className="text-lg font-semibold text-white md:text-2xl">
                {category.title}
              </h3>
            </div>
          </div>
        ))}
      </div>
    </div>
  );
};

export default CategoryGrid;
