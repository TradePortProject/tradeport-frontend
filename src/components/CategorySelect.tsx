import React from 'react';

interface Category {
    id: number;
    name: string;
}

interface CategorySelectProps {
    categories: Category[];
    selectedCategory: number;
    onChange: (selectedCategory: number) => void;
}

const CategorySelect: React.FC<CategorySelectProps> = ({ categories, selectedCategory, onChange }) => {
    const handleChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
        const selectedId = Number(event.target.value);
        onChange(selectedId);
    };

    return (
        <div className="category-select">
            <select
                value={selectedCategory}
                onChange={handleChange}
                className="px-4 py-2 border rounded"
            >
                {categories.map(category => (
                    <option key={category.id} value={category.id}>
                        {category.name}
                    </option>
                ))}
            </select>
        </div>
    );
};

export default CategorySelect;
