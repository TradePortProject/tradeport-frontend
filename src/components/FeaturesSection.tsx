import React from "react";
import {
  TruckIcon,
  CreditCardIcon,
  LockClosedIcon,
  PhoneIcon,
} from "@heroicons/react/24/outline";

interface FeatureProps {
  icon: React.ElementType;
  title: string;
  description: string;
}

const features: FeatureProps[] = [
  { icon: TruckIcon, title: "Free Shipping", description: "Order above $200" },
  {
    icon: CreditCardIcon,
    title: "Money-back",
    description: "30 days guarantee",
  },
  {
    icon: LockClosedIcon,
    title: "Secure Payments",
    description: "Secured by Stripe",
  },
  {
    icon: PhoneIcon,
    title: "24/7 Support",
    description: "Phone and Email support",
  },
];

const FeaturesSection: React.FC = () => {
  return (
    <div className="mx-auto max-w-7xl px-4 py-12">
      <div className="grid grid-cols-2 gap-6 sm:grid-cols-2 md:grid-cols-4">
        {features.map((feature, index) => (
          <div
            key={index}
            className="flex flex-col items-center rounded-lg bg-gray-100 p-6 text-center shadow-sm transition-shadow duration-300 hover:shadow-md"
          >
            <feature.icon className="mb-4 h-10 w-10 text-gray-700" />
            <h3 className="text-lg font-semibold text-gray-900">
              {feature.title}
            </h3>
            <p className="text-sm text-gray-600">{feature.description}</p>
          </div>
        ))}
      </div>
    </div>
  );
};

export default FeaturesSection;
