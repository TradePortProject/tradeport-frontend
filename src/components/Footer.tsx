import React from "react";

import { MagnifyingGlassIcon, EnvelopeIcon } from "@heroicons/react/24/outline";

const Footer: React.FC = () => {
  return (
    <footer className="bg-black py-6 text-white">
      <div className="mx-auto max-w-7xl px-6">
        {/* Top Section: Logo & Navigation */}
        <div className="flex flex-col items-center justify-between border-b border-gray-700 pb-4 md:flex-row">
          {/* Logo & Tagline */}
          <div className="flex items-center space-x-3">
            <span className="text-xl font-semibold">TradePort</span>
            <span className="text-gray-400">| The New Way for B2B</span>
          </div>

          {/* Navigation Links */}
          <nav className="mt-4 flex space-x-6 text-sm text-gray-300 md:mt-0">
            <a
              href="#"
              className="flex items-center space-x-1 transition hover:text-white"
            >
              <span>Home</span>
            </a>
            <a
              href="#"
              className="flex items-center space-x-1 transition hover:text-white"
            >
              <span>Shop</span>
            </a>
            <a
              href="#"
              className="flex items-center space-x-1 transition hover:text-white"
            >
              <span>Product</span>
            </a>
            <a
              href="#"
              className="flex items-center space-x-1 transition hover:text-white"
            >
              <span>Blog</span>
            </a>
            <a
              href="#"
              className="flex items-center space-x-1 transition hover:text-white"
            >
              <span>Contact Us</span>
            </a>
          </nav>
        </div>

        {/* Bottom Section: Copyright & Socials */}
        <div className="flex items-center justify-between pt-4">
          {/* Copyright & Legal Links */}
          <p className="flex text-sm text-gray-400">
            Â© 2025 TradePort. All rights reserved.
            <span className="ml-4">
              <a
                href="#"
                className="flex items-center space-x-1 font-medium transition hover:text-white"
              >
                <span>Privacy Policy</span>
              </a>
            </span>
            <a
              href="#"
              className="ml-2 space-x-1 font-medium transition hover:text-white"
            >
              <span>Terms of Use</span>
            </a>
          </p>

          {/* Social Media Icons */}
          <div className="mt-4 flex space-x-4 md:mt-0">
            <a href="#" className="transition hover:text-gray-300">
              <MagnifyingGlassIcon className="h-6 w-6" />
            </a>
            <a href="#" className="transition hover:text-gray-300">
              <EnvelopeIcon className="h-6 w-6" />
            </a>
          </div>
        </div>
      </div>
    </footer>
  );
};

export default Footer;
