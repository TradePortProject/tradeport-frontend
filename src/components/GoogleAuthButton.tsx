import { GoogleLogin, CredentialResponse } from "@react-oauth/google";
import { useNavigate } from "react-router-dom";
import { login, setUserDetails } from "../store/features/authSlice";
import { useDispatch } from "react-redux";
import axios from "axios";
import { jwtDecode } from "jwt-decode";
import ENDPOINTS from "../config/apiConfig";

const GoogleAuthButton: React.FC = () => {
  const dispatch = useDispatch();
  const navigate = useNavigate();

  const handleSuccess = async (credentialResponse: CredentialResponse) => {
    if (!credentialResponse.credential) {
      console.error("Google authentication failed: No credential received");
      alert("Google login failed. Please try again.");
      return;
    }

    console.log("Received fresh Google Token:", credentialResponse.credential); // Debugging

    try {
      const { email, name, picture } = jwtDecode<{
        email: string;
        name?: string;
        picture?: string;
      }>(credentialResponse.credential);

      dispatch(setUserDetails({ email, name, picture }));

      const response = await axios.post(
        ENDPOINTS.USER.VALIDATE_GOOGLE,
        { token: credentialResponse.credential }, // Send as JSON object
        {
          headers: {
            "Content-Type": "application/json"
          },
          validateStatus: (status: number) => status < 500,
        }
      );

      if (response.status === 200) {
        console.log("Backend validation successful:", response.data);

        // Check if response has nested user object
        const userData = response.data.user
          ? response.data.user
          : response.data;
        const token = response.data.token || credentialResponse.credential;

        console.log("Backend user data:", userData);
        console.log("Type of role:", typeof userData.role);
        console.log("Role value:", userData.role);

        // Create formatted user data from correct location
        const formattedUserData = {
          loginID: userData.email || userData.loginID || "",
          userID: userData.userID || "",
          userName: userData.userName || "",
          role:
            typeof userData.role === "string"
              ? parseInt(userData.role, 10)
              : userData.role,
          phoneNo: userData.phoneNo || "",
          address: userData.address || "",
          remarks: userData.remarks || "",
          isActive: userData.isActive !== undefined ? userData.isActive : true,
        };

        console.log("Formatted user data:", formattedUserData);

        // Dispatch with proper data
        dispatch(
          login({
            token: token,
            user: formattedUserData,
          }),
        );

        // Redirect to profile page
        navigate("/profile");
      } else if (response.status === 401) {
        alert("Invalid Google authentication token. Please try again.");
      } else if (response.status === 404) {
        alert("User not registered. Please register first.");
        navigate("/register");
      } else {
        console.error("Unexpected response:", response);
      }
    } catch (error) {
      console.error("Login Process Failed", error);
      alert("Authentication failed. Please try again.");
    }
  };

  return (
    <div>
      <h3>Login with Google</h3>
      <GoogleLogin
        onSuccess={handleSuccess}
        onError={() => {
          console.error("Google Login Failed");
          alert("Google authentication failed. Please try again.");
        }}
      />
    </div>
  );
};

export default GoogleAuthButton;