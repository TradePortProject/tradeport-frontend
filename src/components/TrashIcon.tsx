import React from "react";

export const TrashIcon = () => {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      className="h-5 w-5"
      viewBox="0 0 20 20"
      fill="currentColor"
    >
      <path
        fillRule="evenodd"
        d="M6 2a1 1 0 00-1 1v1H3a1 1 0 100 2h1v10a2 2 0 002 2h8a2 2 0 002-2V6h1a1 1 0 100-2h-2V3a1 1 0 00-1-1H6zm3 3a1 1 0 112 0v1h2a1 1 0 110 2h-1v7a1 1 0 11-2 0V8H9a1 1 0 110-2h1V5z"
        clipRule="evenodd"
      />
    </svg>
  );
};
