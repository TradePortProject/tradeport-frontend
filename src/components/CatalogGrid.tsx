import React, { useEffect, useState } from 'react';
import { useNavigate } from 'react-router-dom';
import ProductCard from './ProductCard';
import CategorySelect from './CategorySelect';
import PriceRangeSelect from './PriceRangeSelect';
import SearchBar from './SearchBar';
import Pagination from './Pagination';

interface Product {
    productID: string;
    productCode: string;
    manufacturerID: string;
    productName: string;
    description: string;
    category: string; // Updated to match JSON: category as a string
    wholesalePrice: number;
    retailPrice: number;
    retailCurrency: string;
    wholeSaleCurrency: string;
    shippingCost: number;
    quantity: number;
    createdOn: string;
    updatedOn: string;
    isActive: boolean;
    productImage: { productImageURL: string }[]; // Updated to handle array of image URLs
}

const categories = [
    { id: 0, name: "All Categories" },
    { id: 1, name: "Furniture" },
    { id: 2, name: "Fashion" },
    { id: 3, name: "Home & Garden" },
    { id: 4, name: "Computer & Office" },
];

const quantities = [
    { id: 0, name: "All Quantities" },
    { id: 25, name: "Min 25" },
    { id: 50, name: "Min 50" },
    { id: 100, name: "Min 100" },
    { id: 500, name: "Min 500" },
    { id: 1000, name: "Min 1000" },
];

const priceRanges = [
    { label: "All Wholesale Price", min: 0, max: 1000000 },
    { label: "$0 - $50", min: 0, max: 50 },
    { label: "$51 - $100", min: 51, max: 100 },
    { label: "$101 - $200", min: 101, max: 200 },
    { label: "$201 - $500", min: 201, max: 500 },
    { label: "$501 and above", min: 501, max: 1000000 },
];

const retailPriceRanges = [
    { label: "All Retail Price", min: 0, max: 1000000 },
    { label: "$0 - $50", min: 0, max: 50 },
    { label: "$51 - $100", min: 51, max: 100 },
    { label: "$101 - $200", min: 101, max: 200 },
    { label: "$201 - $500", min: 201, max: 500 },
    { label: "$501 and above", min: 501, max: 1000000 },
];

const CatalogGrid: React.FC = () => {
    const [products, setProducts] = useState<Product[]>([]);
    const [selectedCategory, setSelectedCategory] = useState<number>(0);
    const [selectedQuantity, setselectedQuantity] = useState<number>(0);
    const [selectedPrice, setSelectedPriceRange] = useState<string>("All Wholesale Price");
    const [selectedRetailPrice, setSelectedRetailPriceRange] = useState<string>("All Retail Price");
    const [pageNumber, setPageNumber] = useState<number>(1);
    const [totalPages, setTotalPages] = useState<number>(1);
    const [searchText, setSearchText] = useState<string>("");

	useEffect(() => {
		if (searchText.length >= 3 || searchText.length === 0) {
			fetchProducts(); // Call API only if the searchText condition is satisfied
		}
	}, [searchText]);

    useEffect(() => {
        fetchProducts();
    }, [pageNumber, selectedCategory, selectedQuantity, selectedPrice, selectedRetailPrice]);

    const fetchProducts = async () => {
        try {
            const wholesaleRange = getWholesalePriceRange(selectedPrice);
            const retailRange = getRetailPriceRange(selectedRetailPrice);
			const categoryQuery = selectedCategory !== 0 ? `&category=${selectedCategory}` : ' ';
            const response = await fetch(`http://localhost:3016/api/productManagement/GetFilteredProducts?pageNumber=${pageNumber}&searchText=${searchText}${categoryQuery}&quantity=${selectedQuantity}&minWholesalePrice=${wholesaleRange.min}&maxWholesalePrice=${wholesaleRange.max}&minRetailPrice=${retailRange.min}&maxRetailPrice=${retailRange.max}`);
            const data = await response.json();
            console.log('Fetched products:', data.product);
            setProducts(data.product|| []);
            setTotalPages(data.totalPages || 1);
        } catch (error) {
            console.error('Error fetching products:', error);
            setProducts([]); // Clear products on error
        }
    };

    const getWholesalePriceRange = (label: string) => {
        return priceRanges.find(price => price.label === label) || { min: 0, max: 1000000 };
    };

    const getRetailPriceRange = (label: string) => {
        return retailPriceRanges.find(price => price.label === label) || { min: 0, max: 1000000 };
    };

    const handleSearchChange = (newSearchText: string) => {
        setSearchText(newSearchText);
    };
	
    const handleCategoryChange = (selectedId: number) => {
        setSelectedCategory(selectedId);
    };

    const handleQuantityChange = (selectedId: number) => {
        setselectedQuantity(selectedId);
    };

    const handleWholesalePriceRangeChange = (selectedLabel: string) => {
        setSelectedPriceRange(selectedLabel);
    };

    const handleRetailPriceRangeChange = (selectedLabel: string) => {
        setSelectedRetailPriceRange(selectedLabel);
    };

    const handlePageChange = (page: number) => {
        setPageNumber(page);
    };

    return (
        <div className="mx-auto max-w-7xl px-4 py-12">
            <div className="mb-4 flex flex-col md:flex-row justify-between items-center space-y-4 md:space-y-0 md:space-x-4">
                <SearchBar searchText={searchText} onSearchChange={handleSearchChange} />
                <div className="flex flex-col md:flex-row items-center space-y-4 md:space-y-0 md:space-x-4">
                    <CategorySelect categories={categories} selectedCategory={selectedCategory} onChange={handleCategoryChange} />
                    <CategorySelect categories={quantities} selectedCategory={selectedQuantity} onChange={handleQuantityChange} />
                    <PriceRangeSelect priceRanges={priceRanges} selectedPrice={selectedPrice} onChange={handleWholesalePriceRangeChange} />
                    <PriceRangeSelect priceRanges={retailPriceRanges} selectedPrice={selectedRetailPrice} onChange={handleRetailPriceRangeChange} />
                </div>
            </div>
            {products.length > 0 ? (
                <div className="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 gap-6">
                    {products.map(product => (
                        <ProductCard key={product.productID} {...product} />
                    ))}
                </div>
			
            ) : (
                <div>No products found. Adjust your filters.</div>
            )}
			<div className="mt-4 flex justify-center">
				<Pagination pageNumber={pageNumber} totalPages={totalPages} onPageChange={handlePageChange} />
			</div>	
        </div>
    );
};

export default CatalogGrid;
