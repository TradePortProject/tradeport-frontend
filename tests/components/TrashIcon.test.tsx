import { render } from "@testing-library/react";
import { TrashIcon } from "../../src/components/TrashIcon";

describe("TrashIcon", () => {
  it("renders without crashing", () => {
    const { container } = render(<TrashIcon />);
    expect(container.firstChild).toBeInTheDocument();
  });

  // Optionally, you could add a snapshot test
  it("matches snapshot", () => {
    const { container } = render(<TrashIcon />);
    expect(container).toMatchSnapshot();
  });
});
