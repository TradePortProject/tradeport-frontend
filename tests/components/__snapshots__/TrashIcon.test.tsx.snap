// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TrashIcon > matches snapshot 1`] = `
<div>
  <svg
    class="h-5 w-5"
    fill="currentColor"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clip-rule="evenodd"
      d="M6 2a1 1 0 00-1 1v1H3a1 1 0 100 2h1v10a2 2 0 002 2h8a2 2 0 002-2V6h1a1 1 0 100-2h-2V3a1 1 0 00-1-1H6zm3 3a1 1 0 112 0v1h2a1 1 0 110 2h-1v7a1 1 0 11-2 0V8H9a1 1 0 110-2h1V5z"
      fill-rule="evenodd"
    />
  </svg>
</div>
`;
